components/brave_rewards/browser/rewards_p3a.cc

```diff
@@ -180,6 +180,63 @@ double CalcWalletBalance(base::flat_map<std::string, double> wallets,
   return balance_minus_grant;
 }
 
+void RecordRewardsEnabledDuration(PrefService* prefs, bool rewards_enabled) {
+  auto enabled_timestamp = prefs->GetTime(prefs::kEnabledTimestamp);
+  auto value = RewardsEnabledDuration::kNever;
+
+  if (enabled_timestamp.is_null()) {
+    // No previous timestamp, so record one of the non-duration states.
+    if (!rewards_enabled) {
+      // Rewards have been disabled with no previous timestamp.
+      // Probably they've been on since we started measuring.
+      // Ignore this interval since we can't measure it and treat it
+      // the same as never having enabled.
+      value = RewardsEnabledDuration::kNever;
+    } else {
+      // Rewards have been enabled.
+      // Remember when so we can measure the duration on later changes.
+      prefs->SetTime(prefs::kEnabledTimestamp, base::Time::Now());
+      value = RewardsEnabledDuration::kStillEnabled;
+    }
+  } else {
+    // Previous timestamp available.
+    if (!rewards_enabled) {
+      // Rewards have been disabled. Record the duration they were on.
+      // Set the threshold at three units so each bin represents the
+      // nominal value as an order-of-magnitude: more than three days
+      // is a week, more than three weeks is a month, and so on.
+      constexpr auto threshold = 3;
+      constexpr auto days_per_week = 7;
+      constexpr auto days_per_month = 30.44;  // average length
+      auto duration = base::Time::Now() - enabled_timestamp;
```

Comment:
personally I'd not overuse `auto`, e.g. in this case it's convenient to see that `duration` is a TimeDelta
there is a styleguide entry about that https://google.github.io/styleguide/cppguide.html#Type_deduction